src/selectors/Entities/name.js

```diff
@@ -77,16 +79,47 @@ export const selectCanEditPatient = state => {
   return UIDatabase.getPreference(PREFERENCE_KEYS.CAN_EDIT_PATIENTS_FROM_ANY_STORE) || isEditable;
 };
 
-export const selectVaccinePatientHistory = state => {
-  const patientId = selectEditingNameId(state) ?? '';
+const jsonSchema = {
+  type: 'object',
+  properties: {
+    refused: {
+      type: 'boolean',
+      enum: [false],
+    },
+    vaccinator: {
+      type: 'string',
+    },
+    itemName: {
+      type: 'string',
+    },
+    itemCode: {
+      type: 'string',
+    },
+    vaccineDate: {
+      type: 'string',
+    },
+  },
+};
 
-  const inQuery = UIDatabase.objects('Transaction')
-    .filtered('otherParty.id == $0', patientId)
-    .map(({ id }) => `transaction.id == "${id}"`)
-    .join(' OR ');
-  const baseQueryString = 'type != "cash_in" AND type != "cash_out"';
-  const fullQuery = `(${inQuery}) AND ${baseQueryString} AND itemBatch.item.isVaccine == true`;
-  return inQuery ? UIDatabase.objects('TransactionBatch').filtered(fullQuery).slice() : [];
+export const selectVaccinePatientHistory = patient => {
+  const [vaccinationPatientEvent] = UIDatabase.objects('PatientEvent').filtered(
+    "code == 'vaccination'"
+  );
+  const { id: vaccinationPatientEventID } = vaccinationPatientEvent ?? {};
+
+  const nameNotes = patient?.nameNotes
+    ?.filter(
+      ({ patientEventID, data }) =>
+        patientEventID === vaccinationPatientEventID && validateJsonSchemaData(jsonSchema, data)
+    )
+    .map(({ data: vaccinationNameNotes }) => ({
+      ...vaccinationNameNotes,
+      doses: 1, // Currently not possible to dispense more than 1 dose
```

Comment:
There were no quantities in the name_notes created 😅 (maybe we should add one?)

Reply:
Hehe, yeah, we should. For now it's fine to assume `1` 😁  Technically the whole prescription is in there !!! 🤣